### PR TITLE
Implement display of missing key pairs in wallet settings

### LINKED ISSUE #19733: Implement "Missing Keypairs" component
# Feature

Implement the "Missing Keypairs" component from the Wallet category in the quo2 component library.

Missing keypairs

### Designs

[Link to Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=29992%3A23104&mode=design&t=1PM8QIod3cWlObWi-1)

### Acceptance Criteria:

- Should follow the [best practices](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md)
- Should add [component test](https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md)
- Should add an export from the `quo2.core` namespace
- Add a preview for this component under the `wallet` category in the Quo2 preview screen

### LINKED ISSUE #20033: Display missing key pairs in wallet settings
# Feature Issue

### User Story

As a user,
I want to be able to see any of my missing (not imported) key pairs and accounts inside the wallet settings.

### Description 
Implement the initial UI for displaying the missing (not imported) or un-operable key pairs and accounts.

### Designs

Figma flow: [the Figma designs](https://www.figma.com/design/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?node-id=232-32244&t=3gekg5HDrRdVjAye-4)

### Notes

* The designs for importing missing key pairs include an "import" button. This feature is not in scope for this issue and will be tracked in a separate issue.
* The designs for importing missing key pairs include an action menu for each missing key pair. This feature is partially in scope because we already support renaming and removing key pairs. The other features related to importing the key pair will be tracked in a separate issue.

### Acceptance Criteria

| Screen | Description |
|--------|--------|
| Wallet Settings | Users can choose to view the key pairs and accounts settings (inside wallet settings) by pressing on the "Key pairs and accounts" menu button. |
| Missing Key Pairs and Accounts | Users should be able to see any missing key pairs and accounts. Missing key pairs and accounts would exist on other devices but not on the current device. |
| Missing Key Pairs and Accounts Action Menu | Users should be able to open the action menu for a missing key pair. |

### PR DESCRIPTION
**Pull request by seanstrom**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #20033 
fixes #19733

### Summary

* This PR attempts to implement the logic for displaying the missing key-pairs available from a user's account, but not available on the user's device.
* This PR attempts to implement the Quo components for missing-keypairs list and list items.
* This PR attempts to resolve some small bugs with the "Key pairs and accounts" settings screen
* This PR attempts to resolve an issue with Quo standard-title component

#### Platforms

- Android
- iOS

##### Functional

- Mobile wallet settings
- Key pairs and accounts settings

### Steps to test

To test these changes there are some requirements, the user account for testing will need with two devices, and both of those devices must be synced to the user account.

Once the account and devices are setup to sync between the devices:
* Open the Status app (mobile or desktop) on one device and create a new account with a key-pair.
  * Choose to create a new account from the wallet home screen, then select to create a new key-pair by generating a recovery phrase key-pair, and then finish creating the account with that new key-pair.
  * Because the devices are synced, the second device should be updated with the knowledge of the new account and the key-pair.
* Navigate to the mobile wallet setting and open the "Key pairs and accounts" settings.
* At the top of screen there should be a section for missing key-pairs, and that section should display the current missing key-pair and the icon emojis for the related accounts.

### After changes screen capture

https://github.com/status-im/status-mobile/assets/2845768/7a728135-4747-4c9b-af01-47d1ad92fd12

status: ready

### CODE REVIEW

File: src/quo/components/list_items/missing_keypair/view.cljs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+(ns quo.components.list-items.missing-keypair.view
+  (:require
+    [quo.components.avatars.icon-avatar :as icon-avatar]
+    [quo.components.icon :as icon]
+    [quo.components.list-items.missing-keypair.schema :as component-schema]
+    [quo.components.list-items.missing-keypair.style :as style]
+    [quo.components.list-items.preview-list.view :as preview-list]
+    [quo.components.markdown.text :as text]
+    [quo.theme :as quo.theme]
```
Comment (mohsen-ghafouri) [inline src/quo/components/list_items/missing_keypair/view.cljs]:
```suggestion
    [quo.theme]
```

File: src/quo/components/wallet/missing_keypairs/view.cljs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+(ns quo.components.wallet.missing-keypairs.view
+  (:require
+    [quo.components.icon :as icon]
+    [quo.components.list-items.missing-keypair.view :as missing-keypair]
+    [quo.components.markdown.text :as text]
+    [quo.components.wallet.missing-keypairs.style :as style]
+    [quo.foundations.colors :as colors]
+    [quo.theme :as quo.theme]
```
Comment (mohsen-ghafouri) [inline src/quo/components/wallet/missing_keypairs/view.cljs]:
```suggestion
    [quo.theme]
```

**Review by mohsen-ghafouri — APPROVED** [review]
Well done 🔥, LGTM :)

File: src/quo/components/wallet/missing_keypairs/view.cljs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+(ns quo.components.wallet.missing-keypairs.view
```
Comment (smohamedjavid) [inline src/quo/components/wallet/missing_keypairs/view.cljs:1]: Adding previews for this component is highly appreciated
`status-im.contexts.preview.quo.wallet.missing_keypairs`

File: src/status_im/contexts/settings/wallet/keypairs_and_accounts/actions/view.cljs [hunk 4]
```diff
@@ -12,7 +12,7 @@
   [:<>
    [quo/drawer-top props]
    [quo/action-drawer
-    [(when (= (:type props) :keypair)
+    [(when (not= (:type props) :default-keypair)
```
Comment (smohamedjavid) [inline src/status_im/contexts/settings/wallet/keypairs_and_accounts/actions/view.cljs]: 🙌

File: src/status_im/subs/wallet/wallet.cljs [hunk 5]
```diff
@@ -179,21 +179,46 @@
...
-               {:type     (keyword type)
-                :name     name
-                :key-uid  key-uid
-                :accounts (format-settings-keypair-accounts accounts format-options)})))))
-
+ :<- [:wallet/accounts]
+ (fn [[keypairs accounts] [_ format-options]]
+   (let [grouped-accounts      (->> accounts
+                                    (map #(select-keys % [:operable :key-uid]))
+                                    (map #(update-in % [:operable] keyword))
```
Comment (smohamedjavid) [inline src/status_im/subs/wallet/wallet.cljs]: IMO, we should convert this to a keyword when we save it to the app-db. 

https://github.com/status-im/status-mobile/blob/d4e7e4cd52581f4d1db9b916aa9c8ce64aaa1039/src/status_im/contexts/wallet/data_store.cljs#L35-L48

It makes life easier if we need this key in other places.

File: src/status_im/subs/wallet/wallet.cljs [hunk 6]
```diff
@@ -179,21 +179,46 @@
...
-   (->> keypairs
-        (map (fn [{:keys [accounts name type key-uid]}]
-               {:type     (keyword type)
-                :name     name
-                :key-uid  key-uid
-                :accounts (format-settings-keypair-accounts accounts format-options)})))))
-
+ :<- [:wallet/accounts]
+ (fn [[keypairs accounts] [_ format-options]]
+   (let [grouped-accounts      (->> accounts
```
Comment (smohamedjavid) [inline src/status_im/subs/wallet/wallet.cljs:211]: Not a blocker: Just wondering why we didn't use the `accounts` key (an array of accounts) inside keypairs to determine if that keypair is fully operable or not. It would be much simpler that way and we can add that key to the keypair while saving it to the app-db.

I will do some refactoring/cleanups in keypairs data in app-db later 👍

File: src/quo/components/drawers/drawer_top/view.cljs
Comment (smohamedjavid) [inline src/quo/components/drawers/drawer_top/view.cljs:1]: Quick question: I can't find this type `missing-keypair` for the `drawer-top` component in the Design System. If it's not too much trouble, would you help add the link to the design?

**Review by smohamedjavid — COMMENTED** [review]
Nice work 🚀 @seanstrom 🙌 

You might need to add this issue (https://github.com/status-im/status-mobile/issues/19733) to this PR as it's done here. Thanks!

File: src/quo/components/wallet/missing_keypairs/view.cljs (hunk 3 above)
Comment (seanstrom) [reply src/quo/components/wallet/missing_keypairs/view.cljs:1]: Yup yup will do 👍

File: src/status_im/subs/wallet/wallet.cljs (hunk 5 above)
Comment (seanstrom) [reply src/status_im/subs/wallet/wallet.cljs]: Yup yup I agree, I'll update the decoder and update the current usages of `:operable` in the tests to use a keyword.

File: src/status_im/subs/wallet/wallet.cljs (hunk 6 above)
Comment (seanstrom) [reply src/status_im/subs/wallet/wallet.cljs:211]: Initially I hoped to just iterate over all the key-pairs and accounts, but it seemed that key-pairs were not immediately synced with the accounts after creating a new account and key-pair, but accounts seemed to be updated immediately. However, that may have been me discovering a bug (?), because it seems I need to sometimes re-login to re-query the existing key-pairs and accounts.

Maybe I can try to fix that here, or I can try to create a bug issue to investigate. Thoughts?

File: src/quo/components/drawers/drawer_top/view.cljs
Comment (seanstrom) [reply src/quo/components/drawers/drawer_top/view.cljs:1]: Ah yeah good question 🙏 

I don't think it's an official extension of the component, but here's the designs that feature the drawer-top component with a missing key-pair: https://www.figma.com/design/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?node-id=232-33083&t=4nFqEVTIOAcfu0JS-4

Maybe the design system should be updated?

File: src/status_im/subs/wallet/wallet.cljs (hunk 6 above)
Comment (smohamedjavid) [reply src/status_im/subs/wallet/wallet.cljs:211]: I guess I know the problem. From the video in the PR description. The user is not logged in (in Device A - mobile or desktop) and creates a new key pair in other device (Device B - mobile or desktop).

Usually, the sync message (in our case, new key pair details) is received after the user logs in. But, we request the accounts and key pair data as soon as the user logs in and whatever data is available in the status-go is sent back to the client. When the sync message is received (in Device A), it will add those details in the db (in status-go) and notify the client via signal. We need to handle that part.

I will add that support as I'm working on something related to that.

In short, it's not a bug. It's feature needs to be built 😄

Comment (smohamedjavid) [reply src/status_im/subs/wallet/wallet.cljs:211]: The existing code/PR is good to go. We can clean up and refactor later once we handle all those cases (regarding key pairs and accounts). 👍

File: src/quo/components/drawers/drawer_top/view.cljs
Comment (smohamedjavid) [reply src/quo/components/drawers/drawer_top/view.cljs:1]: The type here is still `keypair` for the drawer-top component. I believe it's just the text (subtitle in our code) change that it needs to receive from the parent instead of hardcoding in the component.  

We should confirm with the design team.

Comment (seanstrom) [reply src/quo/components/drawers/drawer_top/view.cljs:1]: Okay I'll double check with design team 👍

Comment (seanstrom) [reply src/quo/components/drawers/drawer_top/view.cljs:1]: Still double checking with the design team, but for the moment I've refactored drawer-top component to use a `:stored` field (similar to the `quo/keypair` component) for describing whether the key pair is `:on-device`, `:on-keycard`, or `:missing`.

Comment (smohamedjavid) [reply src/quo/components/drawers/drawer_top/view.cljs:1]: It should be fine as we have the same in the `quo/keypair` component. Let's post this information to the Design team to keep them updated. 👍

**Review by smohamedjavid — APPROVED** [review]
🚀

**Comment by seanstrom** [comment]
@status-im/mobile-qa can you review these E2E results please? 🙏

**Comment by yevh-berdnyk** [comment]
> @status-im/mobile-qa can you review these E2E results please? 🙏

Hi @seanstrom, thanks for the PR! E2E failures are not related

**Comment by seanstrom** [comment]
Note, this PR is going to use `skip-manual-qa` because most of the work is feature flagged.
We'll ideally be doing a design review once the feature flag for key pairs and accounts is removed.
